config/opal_setup_wrappers.m4

```diff
@@ -220,18 +230,19 @@ EOF
 AC_DEFUN([OPAL_SETUP_RUNPATH],[
     OPAL_VAR_SCOPE_PUSH([LDFLAGS_save rpath_script rpath_outfile wl_fc])
 
-    AC_MSG_CHECKING([if linker supports RUNPATH])
     # Set the output in $runpath_args
     runpath_args=
     LDFLAGS_save=$LDFLAGS
     LDFLAGS="$LDFLAGS -Wl,--enable-new-dtags"
-    AC_LANG_PUSH([C])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([], [return 7;])],
-                   [WRAPPER_RPATH_SUPPORT=runpath
-                    runpath_args="-Wl,--enable-new-dtags"
-                    AC_MSG_RESULT([yes (-Wl,--enable-new-dtags)])],
-                   [AC_MSG_RESULT([no])])
-    AC_LANG_POP([C])
+    AS_IF([test x"$enable_wrapper_runpath" = x"yes"],
+           [AC_LANG_PUSH([C])
+            AC_MSG_CHECKING([if linker supports RUNPATH])
+            AC_LINK_IFELSE([AC_LANG_PROGRAM([], [return 7;])],
+                           [WRAPPER_RPATH_SUPPORT=runpath
+                            runpath_args="-Wl,--enable-new-dtags"
```

Comment:
This doesn't seem right.  Can you please double check this in the context of the rest of the configure script?

Specifically: `$runpath_args` is used later down in the code, but you've now only assigned it if `--disable-wrapper-runpath` was not specified.

Are the wrappers the *only* place that runpath is used?

Reply:
hmmm looks like we may need some other commits from master, I'm surprised the cherry-pick worked without a problem.

Reply:
actually runpath_args  is initialized empty at line 234.  the variable is used at line 315.  The admin wanting the patch on 3.0.x noticed an issue with the fortran wrappers not obeying the --enable/disable-runpath, but I'll fix that in a separate PR.

Reply:
I was wrong.  the patch looks correct. see comments on the file.

Reply:
Right, I see it's initialized in 234.  But it's *used* down below.  My question is: are there any other places we use runpath besides the wrappers?  Because if so, this patch is disabling runpath *everywhere*, not just for the wrappers.

If we're only using runpath for the wrappers, then this patch is likely fine.

Reply:
It doesn't look like it.  buildrpm.sh does add --enable-new-dtags to LDFLAGS passed into configure for the spec file.  Looks like this change has been in master for almost 2 years now.